src/Core/Checkout/Document/Zugferd/ZugferdBuilder.php

```diff
@@ -0,0 +1,91 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Document\Zugferd;
+
+use DateTimeInterface;
+use horstoeko\zugferd\ZugferdDocumentBuilder;
+use horstoeko\zugferd\ZugferdProfiles;
+use Shopware\Core\Checkout\Cart\LineItem\LineItem;
+use Shopware\Core\Checkout\Cart\Price\Struct\CartPrice;
+use Shopware\Core\Checkout\Document\DocumentConfiguration;
+use Shopware\Core\Checkout\Document\DocumentException;
+use Shopware\Core\Checkout\Document\Struct\DocumentGenerateOperation;
+use Shopware\Core\Checkout\Order\Aggregate\OrderAddress\OrderAddressEntity;
+use Shopware\Core\Checkout\Order\Aggregate\OrderCustomer\OrderCustomerEntity;
+use Shopware\Core\Checkout\Order\Aggregate\OrderDelivery\OrderDeliveryCollection;
+use Shopware\Core\Checkout\Order\Aggregate\OrderLineItem\OrderLineItemCollection;
+use Shopware\Core\Checkout\Order\Aggregate\OrderLineItem\OrderLineItemEntity;
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+#[Package('checkout')]
+class ZugferdBuilder
+{
+    /**
+     * @internal
+     */
+    public function __construct(
+        protected EventDispatcherInterface $eventDispatcher
+    ) {
+    }
+
+    public function buildDocument(OrderEntity $order, DocumentGenerateOperation $operation, DocumentConfiguration $config, Context $context): string
+    {
+        $config->__set('isGross',  match ($order->getTaxStatus()) {
```

Comment:
can't you just use `setIsGross()`? This is a magic setter and should function this way. Or add setters / getters to the config.